src/libs/extensionsystem/pluginspec.cpp

```diff
@@ -209,6 +215,19 @@ QString PluginSpec::name() const
     return d->name;
 }
 
+/*!
+    \fn QString PluginSpec::shortName() const
+    The short plugin name. This is valid after the PluginSpec::Read state is reached.
+
+    \note This property is only available on OS/2.
+*/
+#ifdef  Q_OS_OS2
+QString PluginSpec::shortName() const
+{
+    return d->shortName;
+}
+#endif
+
```

Comment:
As discussed on Slack, is it really necessary to introduce all this short name burden on the Qt Creator source level? Plugins are dynamically loaded, so we have full control over name resolution. In Qt itself (in QML) I do it like that:
1. Build a plugin DLL with a short name as required by the OS/2 kernel.
2. Create a plugin symlink with a normal (long) name that points to it.
3. `QLibrary` on OS/2 uses `dlopen` instead of `DosLoadModule` which, besides adding fork support, also handles symlink resolution.

I did that this way for QML plugins because they are referenced by full name from `qmldir` files and fixing that would be a major PITA. Looks like Qt Creator is the same case.

Please see https://github.com/bitwiseworks/qtdeclarative-os2/blob/master/src/imports/layouts/layouts.pro as an example. It gives this result in the build tree:
```
drwxr-xr-x 1 root root       0 Mar 17 00:22 .
drwxr-xr-x 1 root root       0 Jan 16 19:10 ..
-rw-r--r-- 1 root root    3689 Sep  2  2019 plugins.qmltypes
-rw-r--r-- 1 root root     125 Sep  2  2019 qmldir
-rw-rw-rw- 1 root root 1518510 Jan 16 19:12 qqlp.dll
-rw-rw-rw- 1 root root 1883200 Mar 17 00:22 qqlpd.dll
lrwxrwxrwx 1 root root       8 Jan 16 19:12 qquicklayoutsplugin.dll -> qqlp.dll
lrwxrwxrwx 1 root root       9 Mar 17 00:22 qquicklayoutsplugind.dll -> qqlpd.dll
```
The "magic" that does symlink creation is dragged in by `include(qml_plugin)` in the .pro file. Look at https://github.com/bitwiseworks/qtbase-os2/blob/master/mkspecs/features/qml_plugin.prf for details (search for "short" and "os2") there.